game/src/main/kotlin/content/skill/summoning/ShardSwapping.kts

```diff
@@ -0,0 +1,199 @@
+package content.skill.summoning
+
+import content.entity.player.bank.noted
+import content.entity.player.dialogue.type.intEntry
+import world.gregs.voidps.cache.definition.data.InterfaceDefinition
+import world.gregs.voidps.engine.client.message
+import world.gregs.voidps.engine.client.sendScript
+import world.gregs.voidps.engine.client.ui.interfaceOption
+import world.gregs.voidps.engine.client.ui.open
+import world.gregs.voidps.engine.data.definition.EnumDefinitions
+import world.gregs.voidps.engine.data.definition.ItemDefinitions
+import world.gregs.voidps.engine.entity.character.npc.npcOperate
+import world.gregs.voidps.engine.entity.character.player.Player
+import world.gregs.voidps.engine.entity.item.Item
+import world.gregs.voidps.engine.inject
+import world.gregs.voidps.engine.inv.inventory
+import world.gregs.voidps.engine.inv.transact.operation.AddItem.add
+import world.gregs.voidps.engine.inv.transact.operation.RemoveItem.remove
+import kotlin.math.min
+
+val enums: EnumDefinitions by inject()
+val itemDefinitions: ItemDefinitions by inject()
+
+npcOperate("Swap", "bogrog") {
+    openTradeInInterface(player, true)
+}
+
+interfaceOption("Trade Scrolls", id = "summoning_trade_in") {
+    openTradeInInterface(player, false)
+}
+
+interfaceOption("Trade Pouches", id = "summoning_trade_in") {
+    openTradeInInterface(player, true)
+}
+
+interfaceOption("Value", "*_trade_in", "summoning_trade_in") {
+    val enumIndex = (itemSlot + 3) / 5
+    var actualItem = item
+    val itemType = component.removeSuffix("_trade_in")
+
+    if (item.id.endsWith("_u")) {
+        val actualItemId = enums.get("summoning_${itemType}_ids_1").getInt(enumIndex)
+        actualItem = Item(itemDefinitions.get(actualItemId).stringId)
+    }
+
+    sendValueMessage(player, actualItem, itemType)
+}
+
+interfaceOption("Trade*", "*_trade_in", "summoning_trade_in") {
+    val enumIndex = (itemSlot + 3) / 5
+    var actualItem = item
+    val itemType = component.removeSuffix("_trade_in")
+
+    if (item.id.endsWith("_u")) {
+        val actualItemId = enums.get("summoning_${itemType}_ids_1").getInt(enumIndex)
+        actualItem = Item(itemDefinitions.get(actualItemId).stringId)
+        sendValueMessage(player, actualItem, itemType)
+        return@interfaceOption
+    }
+
+    when(option) {
+        "Trade" -> swapForShards(player, actualItem, 1)
```

Comment:
Trading in 1 scroll when the required scroll count > 1 doesn't work for some reason 🤔